ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -289,11 +329,14 @@ private PullQueryResult handlePullQuery(
     // increasing order of lag.
     for (KsqlNode node : filteredAndOrderedNodes) {
       try {
+        final TableRows rows
+            = routeQuery(node, statement, executionContext, serviceContext, pullQueryContext);
         final Optional<KsqlNode> debugNode = Optional.ofNullable(
             routingOptions.isDebugRequest() ? node : null);
-        return new PullQueryResult(
-            routeQuery(node, statement, executionContext, serviceContext, pullQueryContext),
-            debugNode);
+        final List<Optional<KsqlNode>> debugNodes = rows.getRows().stream()
```

Comment:
nit: probably makes sense to make this `Optional<List<KsqlNode>>` instead of `List<Optional<KsqlNode>>` to prevent creating a list of empty optionals in the non-debug case. And then you can also use [`Collections#nCopies`](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#nCopies(int,%20T)) to make it a little more readable

Reply:
I had been thinking that maybe this was necessary because different machines can have different configs, but I now remember that this is a request config, so it should always work.  Ok, made it `Optional<List<KsqlNode>>`